public/app/features/query/components/AppLinks.tsx

```diff
@@ -0,0 +1,99 @@
+import { css } from '@emotion/css';
+import { first } from 'lodash';
+import { useEffect, useState } from 'react';
+
+import { DataQuery, PluginExtensionPoints, PluginMeta } from '@grafana/data';
+import { GrafanaTheme2 } from '@grafana/data/';
+import { QueryToAppPluginContext } from '@grafana/data/src/types/pluginExtensions';
+import { usePluginLinks } from '@grafana/runtime';
+import { Dropdown, LinkButton, ToolbarButton } from '@grafana/ui';
+import { useStyles2, Tooltip } from '@grafana/ui/';
+
+import { Trans } from '../../../core/internationalization';
+import { ToolbarExtensionPointMenu } from '../../explore/extensions/ToolbarExtensionPointMenu';
+import { getPluginMeta } from '../../plugins/admin/api';
+
+type Props = {
+  query: DataQuery;
+};
+
+function getStyles(theme: GrafanaTheme2) {
+  return {
+    goQueryLessLink: css({
+      marginRight: '3px',
+      height: '20px',
+      paddingLeft: '5px',
+      paddingRight: '5px',
+      fontSize: theme.typography.bodySmall.fontSize,
+    }),
+    goQueryLessIcon: css({
+      marginLeft: '5px',
+      marginTop: '-1px',
+      width: '11px',
+    }),
+  };
+}
+
+export const AppLinks = ({ query }: Props) => {
+  const context: QueryToAppPluginContext = {
+    query,
+    datasource: query.datasource,
+  };
+
+  const [isOpen, setIsOpen] = useState<boolean>(false);
+  const [plugins, setPlugins] = useState<Record<string, PluginMeta> | undefined>(undefined);
+
+  let links = usePluginLinks({
+    extensionPointId: PluginExtensionPoints.QueryToAppPlugin,
+    context,
+  }).links;
+
+  const styles = useStyles2(getStyles);
+
+  useEffect(() => {
+    links.forEach((link) => {
+      getPluginMeta(link.pluginId).then((meta) => {
```

Comment:
Hmm, I totally understand the pain of getting the plugin meta data async here.

Correct me if I am wrong, but you only need the small logo from the metadata, right? If yes, maybe it would make sense to extend the metadata that we have in the `{ config } from "@grafana/runtime"` instead? (Which is basically coming from `window.grafanaBootData`).

I think we could update the [`newAppDTO()`](https://github.com/grafana/grafana/blob/189802d3c3e0aa876d148e1f707b44afd5e3a563/pkg/api/frontendsettings.go#L564), [the AppDTO struct](https://github.com/grafana/grafana/blob/0db65d229e36b78802c1e8bd0713ac44e7a7cdc7/pkg/plugins/models.go#L321) and [the `AppPluginConfig`](https://github.com/grafana/grafana/blob/0fae3579e88fd5c044bf1688baadce6e6aec3dbc/packages/grafana-runtime/src/config.ts#L40), and then we could get hold of the logos for an app like:

```ts
import { config } from "@grafana/runtime";

// ...

console.log(config.apps[pluginId]?.logos?.small);

```

Reply:
Sorry if I'm late to the party here, but what's the rationale for showing the plugin's logo? Remember one plugin could register multiple extensions to this extension point.  

When registering an extension, you can pass an [icon](https://github.com/grafana/grafana/blob/86bc087257ddf643a693899822669f87d50e1202/packages/grafana-data/src/types/pluginExtensions.ts#L30). Maybe better to use that pattern rather than resolving the link over here?

Reply:
I assume the icon name needs to be from icon bundle which may not have the app icon there?

Reply:
> Sorry if I'm late to the party here, but what's the rationale for showing the plugin's logo? Remember one plugin could register multiple extensions to this extension point.

> When registering an extension, you can pass an [icon](https://github.com/grafana/grafana/blob/86bc087257ddf643a693899822669f87d50e1202/packages/grafana-data/src/types/pluginExtensions.ts#L30). Maybe better to use that pattern rather than resolving the icon over here?

Showing an icon was based on [the feedback](https://raintank-corp.slack.com/archives/C075BDBTX96/p1728053473859169). The icon in the ExtensionLink is `icon?: IconName;` but here it's the link of the app, not `IconName`.